sbi/inference/posteriors/base_posterior.py

```diff
@@ -582,6 +591,139 @@ def sample_conditional(
 
         return samples.reshape((*sample_shape, -1))
 
+    def map(
+        self,
+        x: Optional[Tensor] = None,
+        num_iter: int = 1000,
```

Comment:
is this a reasonable default? 
Maybe we should use a default here at all? It probably depends on lot on the dimensionality of the problem, no? 

Was there a big difference when you tested it on say a 30D problem vs a 2D problem?

Reply:
It was surprisingly similar for my 31D problem vs the 2D problem. It converged quite well after 600 iterations for 31D and after 400 iterations for 3D. So I guessed 1000 was reasonable.